tests/eas/load_tracking.py

```diff
@@ -1273,3 +1290,224 @@ def test_util_tg4_group(self):
+        }
+
+    @classmethod
+    def setUpClass(cls, *args, **kwargs):
+        super(CgroupsMigrationTest, cls).runExperiments(*args, **kwargs)
+
+    @classmethod
+    def _migrate_task(cls, test_env):
+        exclude = ''
+        cgroups = test_env.target.cgroups
+
+        tg1_tasks = cgroups.controllers['cpu'].tasks('/tg1')
+        for pid, (task_name, cmdline) in tg1_tasks.iteritems():
+            # Get the PID of the task t1_1 which is not migrated
+            if task_name == 'rt-app' and re.search(r't1_1', cmdline):
+                exclude = '{}'.format(pid)
+                break
+
+        # Migrate t1_2 to the group /tg2 without migrating t1_1
+        task = cls.root_group.get_child('t1_2')
+        new_taskgroup = cls.root_group.get_child('/tg2')
+        task.change_taskgroup(new_taskgroup)
+        srcg = cgroups.controllers['cpu'].cgroup('/tg1').directory
+        dstg = cgroups.controllers['cpu'].cgroup('/tg2').directory
+        cgroups.cgroups_tasks_move(srcg, dstg, exclude=exclude)
+
+    def test_util_root_group(self):
+        """Test utilization propagation to cgroup root"""
+        return self._test_group_util('/')
+
+    def test_util_tg1_group(self):
+        """Test utilization propagation to cgroup /tg1"""
+        return self._test_group_util('/tg1')
+
+    def test_util_tg2_group(self):
+        """Test utilization propagation to cgroup /tg2"""
+        return self._test_group_util('/tg2')
+
+class NestedCgroupsMigrationTest(_PELTTaskGroupsTest):
+    """
+    Test PELT utilization for task groups migration.
+    Initial group hierarchy:
+                            +-----+
+                            | "/" |
+                            +-----+
+                             /
+                            /
+                       +------+
+                       |"/tg1"|
+                       +------+
+                         /
+                        /
+                +----------+
+                |"/tg1/tg2"|
+                +----------+
+                   /    \
+                  /      \
+                t2_1    t2_2
+
+    Final group hierarchy:
+                            +-----+
+                            | "/" |
+                            +-----+
+                             /
+                            /
+                       +------+
+                       |"/tg1"|
+                       +------+
+                         /   \
+                        /     \
+                +----------+ t2_2
+                |"/tg1/tg2"|
+                +----------+
+                   /
+                  /
+                t2_1
+    """
+    target_cpu = 0
+    root_group = None
+    trace = None
+
+    @classmethod
+    def _getExperimentsConf(cls, test_env):
+        # Run all workloads on a CPU with highest capacity
+        cls.target_cpu = [min(test_env.calibration(),
+                              key=test_env.calibration().get)]
+        period_ms = 16
```

Comment:
Cannot these two be class attributes?

Reply:
Done for period_ms and duration_s (but cls.target_cpu need to be in the function because test_env is only available there)